src/vs/workbench/services/telemetry/browser/telemetryService.ts

```diff
@@ -38,6 +38,34 @@ export class TelemetryService extends Disposable implements ITelemetryService {
 		super();
 
 		if (supportsTelemetry(productService, environmentService) && productService.aiConfig?.ariaKey) {
+			this.impl = this.initializeService(environmentService, loggerService, configurationService, storageService, productService, remoteAgentService);
+		} else {
+			this.impl = NullTelemetryService;
+		}
+
+		// When the level changes it could change from off to on and we want to make sure telemetry is properly intialized
+		this._register(configurationService.onDidChangeConfiguration(e => {
+			if (e.affectsConfiguration(TELEMETRY_SETTING_ID)) {
+				this.impl = this.initializeService(environmentService, loggerService, configurationService, storageService, productService, remoteAgentService);
+			}
+		}));
+	}
+
+	/**
+	 * Initializes the telemetry service to be a full fledged service.
+	 * This is only done once and only when telemetry is enabled as this will also ping the endpoint to
+	 * ensure its not adblocked and we can send telemetry
+	 */
+	private initializeService(
+		environmentService: IBrowserWorkbenchEnvironmentService,
+		loggerService: ILoggerService,
+		configurationService: IConfigurationService,
+		storageService: IStorageService,
+		productService: IProductService,
+		remoteAgentService: IRemoteAgentService
+	) {
+		const telemetrySupported = supportsTelemetry(productService, environmentService) && productService.aiConfig?.ariaKey;
+		if (telemetrySupported && this.impl === NullTelemetryService && this.telemetryLevel.value !== TelemetryLevel.NONE) {
```

Comment:
@lramos15 Should not it be `this.impl === undefined || this.impl === NullTelemetryService` otherwise it always goes to NullTelemetryService now initially even if `this.telemetryLevel.value` is not NONE

Reply:
actually not, this.telemetryLevel depends on this.impl too

Reply:
`NullTelemetryService` is just a no-op black hole. So when that is initialized nothing is initialized or sent where the full fledged BaseTelemetryService respects the settings but sometimes sends health pings etc to the endpoint to make sure it's online which when you have turned telemetry off we don't want to happen. Therefore whenever the setting is completely off we swap in this `NullTelemtryService`. `this.impl` is just the base telemetry layer we're using here as web adds some syntactic sugar on top.

Reply:
@lramos15 actually @akosyakov is right telemetry is broken in vscode.dev insiders, created a PR fixing it https://github.com/microsoft/vscode/pull/159712, probably want to merge it before release :slightly_smiling_face: 
![telemetry_bug](https://user-images.githubusercontent.com/25115070/187789679-8be43bae-b808-4e9f-889a-45b7069d9996.gif)